repository/src/raw.rs

```diff
@@ -101,6 +106,7 @@ pub trait RawRepository: Send + Sync + 'static {
         &mut self,
         commit_message: &str,
         diff: Option<&str>,
+        branch: &Branch,
```

Comment:
Why is this field added? btw I've seen the TODO commit that it will be removed. Just curious.

Reply:
create_commit function is not implemented as it is intended to use "diff" now and knowing current branch is not implemented well now, so I just put it and it will be removed after I implement it correctly.

Reply:
understood.

Reply:
Btw how is it supposed to get currently checked out branch? Did I miss the interface for it?

Reply:
Fix it by using get_head() which gives a commit hash of currently checked out branch. Please see new commit.